### ISSUE TITLE
False positive with rule consistent-this

### ISSUE THREAD

**Issue by webgefrickel** [eslint/eslint#2669]
I have the eslint rule consistent-this enabled in .eslintrc via:

```
"consistent-this": [ 1, "that" ],
```

For the following code snippet, i get a warning: "Unexpected alias 'sum' for 'this'  consistent-this":

```js
var $ = require('jquery');
var total = [ 1, 2, 3 ];
var sum = 0;

$.each(total, function () {
  sum += this;
});

console.log(sum);
```

I am not trying to assign this to anything, just trying to get the sum of all items in the total-array. (I know there are other ways to do that (without jQuery), nevertheless, there should no warning be thrown, if I understand this rule correctly — if not: enlighten me). Thank you!

Edit: running 0.22.1

**Comment by nzakas** [comment]
Working as expected. This is still an assignment, and so the rule is correctly warning you about it. You can use `// eslint-disable consistent-this` to disable around just that instance if you'd like.

**Comment by webgefrickel** [comment]
Thanks a lot for pointing me to // eslint-disable!
